### PR TITLE
Use gravatar URL if found when registering user

### LINKED ISSUE #161: Use Gravatar to automatically fetch avatar
I suspect that this was the original intention, but it would be nice if a gravatar URL could be fetched at registration from the user's email address (if found).

### PR DESCRIPTION
**Pull request by ewels**
Closes #161 

NB: Not entirely sure what happens if running totally offline. I couldn't test this easily as I need a web connection for the SMTP email registration.

### CODE REVIEW

**Comment by pditommaso** [comment]
Hey man, you looks an excellent Groovy developer :)

File: tower-backend/src/main/groovy/io/seqera/tower/service/UserServiceImpl.groovy [hunk 1]
```diff
@@ -143,6 +143,23 @@ class UserServiceImpl implements UserService {
         UserRole userRole = new UserRole(user: user, role: role)
         userRole.save()
 
+        // Try to get a gravatar avatar URL
+        try {
+            String emailHash = email.trim().toLowerCase().md5()
+            String gravatarURL = "https://www.gravatar.com/avatar/${emailHash}"
+            def gravatarConnection = new URL("${gravatarURL}?d=404").openConnection().with {
+                requestMethod = 'HEAD'
```
Comment (pditommaso) [inline tower-backend/src/main/groovy/io/seqera/tower/service/UserServiceImpl.groovy]: What is use for `d` parameter? image size?

Comment (ewels) [reply tower-backend/src/main/groovy/io/seqera/tower/service/UserServiceImpl.groovy]: No - by default, gravatar returns a generic image if the email isn't found:

default

Adding the `?d=404` tells it to give a 404 file not found error instead, if the email isn't found.

https://en.gravatar.com/site/implement/images/

**Comment by ewels** [comment]
> Hey man, you look an excellent Groovy developer :)

I was kind of amazed how easy this was actually! A quick search for `avatar`, one or two google searches and it worked first time! I guess is such a thing as beginner's luck! 😅

File: tower-backend/src/main/groovy/io/seqera/tower/service/UserServiceImpl.groovy (hunk 1 above)
Comment (ewels) [reply tower-backend/src/main/groovy/io/seqera/tower/service/UserServiceImpl.groovy]: ..you can specify image size with `s=[width-in-px]` though.

**Comment by ewels** [comment]
> NB: Not entirely sure what happens if running totally offline. I couldn't test this easily as I need a web connection for the SMTP email registration.

I added a catch-all block around this so hopefully it shouldn't crash the server. It's a bit crude though.

**Comment by pditommaso** [comment]
Nice. I've refactored your code to use the [Micronaut http client](https://docs.micronaut.io/latest/guide/index.html#httpClient) and added a unit test. I'm going to merge it manually.

**Comment by pditommaso** [comment]
Done! 🎉

**Comment by ewels** [comment]
Nice! 👨‍🎤

**Pull request closed without merging**